File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<TbBrand> findAll();

    public PageResult findPage(int pageNum,int pageSize);

    void insert(TbBrand tbBrand);

    void update(TbBrand tbBrand);

    TbBrand findOne(long id);

    void delete(long[] ids);

    PageResult findPage(TbBrand tbBrand, int page, int size);

    public List<Map> selectOptionList();

}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//创建服务
app.service("brandService",function ($http) {
    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };
    //分页查询
    this.findPage = function (page,size) {
        return $http.get('../brand/findPage.do?page='+page+'&size='+size)
    };
    //查询一个
    this.findOne = function (id) {
        return $http.post('../brand/findOne.do?id='+id);
    };
    //添加
    this.add = function (brand) {
        return $http.post('../brand/insert.do',brand)
    };
    //修改
    this.update = function (brand) {
        return $http.post('../brand/update.do',brand)
    };
    //删除
    this.delete = function (selectIds) {
        return $http.get('../brand/delete.do?ids=' + selectIds)
    };
    //条件查询
    this.search = function (page,size,searchBrand) {
        return $http.post('../brand/search.do?page='+page+'&size='+size,searchBrand)
    }
    //查询下拉列表数据
    this.selectOptionList = function(){
        return $http.get('../brand/selectOptionList.do');
    }

});